src/poetry/installation/executor.py

```diff
@@ -104,6 +104,10 @@ def __init__(
         self._shutdown = False
         self._hashes: dict[str, str] = {}
 
+        # Cache whether decorated output is supported.
+        # https://github.com/python-poetry/poetry/issues/9334
```

Comment:
Could you switch the linked issue to the one in cleo?

Reply:
sure, i can even remove the comment and link to the issue in the commit itself so as to not clutter the code if that's preferred?

Reply:
I would prefer it as a comment in the code. It's more visible that way

Reply:
changed this and rebased on main

Reply:
resolving this per approval